src/compiler/parser.ts

```diff
@@ -5616,6 +5617,7 @@ namespace ts {
                         return parseTokenNode<JSDocType>();
                 }
 
+                // TODO (drosen): Parse string literal types in JSDoc as well.

```

Comment:
Not sure we'd need to do this - strings types aren't directly valid [jsdoc types](https://developers.google.com/closure/compiler/docs/js-for-compiler#types). They'd have to be wrapped up inside an enum, 'a la:
```js
/**
* @enum
*/
var MyEnum = {
  Member: "Member"
}

/**
* @param {MyEnum} param
*/
function foo(param) {

}
```
So literal string types should never appear in JSDoc comments, since nothing really supports them.